File: javascript/src/step/count.js
import { Lead } from "../lead.js";

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

export async function count(state) {
    console.log(`Current count: ${state}`);
    await sleep(1000);
    state++;
    if (Math.random() > 0.5) {
        return [state, Lead.Count];
    } else {
        return [state, Lead.Stop];
    }
}

File: go/pkg/step/count.go
package step

import (
	"fmt"
	"math/rand"
	"time"

	"github.com/argenkiwi/ambler/go/pkg/lead"
)

func Count(state int) (int, lead.Lead, error) {
	fmt.Printf("Current count: %d\n", state)
	time.Sleep(1 * time.Second)
	state++
	if rand.Float64() > 0.5 {
		return state, lead.Count, nil
	} else {
		return state, lead.Stop, nil
	}
}
